lib/liquid/combinators/general.ex

```diff
@@ -202,8 +202,19 @@ defmodule Liquid.Combinators.General do
       string(@start_tag)
     ])
     |> reduce({List, :to_string, []})
+    |> traverse({Liquid.Combinators.General, :check_empty, []})
   end
 
+  def check_empty(_rest, args, context, _line, _offset) do
+    case is_empty?(args) do
+      true -> {:error, "Not a valid literal"}
+      false -> {args, context}
+    end
+  end
+
+  defp is_empty?([""]), do: true
```

Comment:
You can delete `is_empty?`. However remember, in elixir you don't need use `is`, the question mark is enough to remark the Boolean nature of a function. You don't use `is_even?` only `even?`.